// Connection message for udaconnect app
// Compile with:
// python3 -m grpc_tools.protoc -I./ --python_out=./ --grpc_python_out=./ connection.proto
syntax = "proto3";

import "person.proto";

//message Person {
//
//  int32 id = 1;
//  string first_name = 2;
//  string last_name = 3;
//  string company_name = 4;

//}

message Location {

  int32 id = 1;
  int32 person_id = 2;
  float longitude = 3;
  float latitude = 4;
  float creation_time = 5;

}

message Connection {

  Person person = 1;
  Location location = 2;

}

message ConnectionRequest {

  int32 person_id = 1;
  string start_date = 2;
  string end_date = 3;
  int32 meters = 4;

} 

message ConnectionData {

  repeated Connection connections = 1;

}

service ConnectionService {

  rpc Get(ConnectionRequest) returns (ConnectionData);

}
